### PR TITLE
[RFC] Dockerfiles: Use busybox:glibc as base image for default image.

### PR DESCRIPTION
**Pull request by eiffel-fl**

### CODE REVIEW

File: Dockerfiles/gadget-core.Dockerfile [hunk 1]
```diff
@@ -42,26 +42,18 @@ RUN cd /gadget/gadget-container && \
...
 	rm -f /etc/localtime && ln -sf /host/etc/localtime /etc/localtime
 
 COPY gadget-container/entrypoint.sh gadget-container/cleanup.sh /
 
 COPY --from=builder /gadget/gadget-container/bin/gadgettracermanager /bin/
 
+# We need to copy libseccomp.so and libdl.so from builder image as
+# gadgettracermanager relies on them.
+COPY --from=builder /usr/lib/x86_64-linux-gnu/libseccomp.so.2 /lib/x86_64-linux-gnu/libdl.so.2 /lib
+
```
Comment (mauriciovasquezbernal) [inline Dockerfiles/gadget-core.Dockerfile:56]: Is this safe? Can we always use this regardless of the base image we use to build our container?

Comment (alban) [reply Dockerfiles/gadget-core.Dockerfile:56]: I think it is not safe in general: `*.so` files can be compiled with different options across-distros so they might not have the same dependencies.

@eiffel-fl Did you have a specific issue with DNS in the gadget-core image? With wget in entrypoint.sh downloading from btfhub?

File: Dockerfiles/gadget-core.Dockerfile [hunk 2]
```diff
@@ -42,26 +42,18 @@ RUN cd /gadget/gadget-container && \
...
-	if command -v tdnf; then \
-		tdnf install -y libseccomp wget util-linux; \
-	elif command -v yum; then \
-		yum install -y libseccomp wget util-linux; \
-	elif command -v apt-get; then \
-		apt-get update && \
-		apt-get install -y seccomp wget util-linux; \
-	elif command -v apk; then \
-		apk add gcompat libseccomp wget util-linux; \
-	fi && \
```
Comment (mauriciovasquezbernal) [inline Dockerfiles/gadget-core.Dockerfile:57]: I think it'll break if a base image that doesn't provide the utilities needed in entrypoint.sh is used (#801).

**Review by mauriciovasquezbernal — COMMENTED** [review]
I'm not sure if we should change this right now. Inspektor Gadget doesn't make a heavy use of DNS: (1) to download some files at startup on entrypoint.sh and (2) to contact the kubeapi-server, hence the lack of DNS over TCP support doesn't seem too relevant for us. Also, you mentioned this is being implemented...

I'd rather take the opportunity to discuss about distroless, once https://github.com/inspektor-gadget/inspektor-gadget/issues/801 is done, can we go distroless? Do we want to do it or is it better to have some utilities (like a package manager) available on our image?

**Comment by alban** [comment]
In general, I am fine with switching to busybox if it works.

It removes socat from the image (`apt-get install -y util-linux socat` line 49), so this PR needs to be on hold until we have an alternative solution to socat.

Current error:

> time="2023-09-25T09:21:11Z" level=warning msg="connecting to gadget service on node \"minikube-containerd-m02\": Internal error occurred: error executing command in container: failed to exec in container: failed to start exec \"1258a9ac460cc719019e70954fcfd4968008c378b325706f9154594cd659dd3e\": OCI runtime exec failed: exec failed: unable to start container process: exec: \"/usr/bin/socat\": stat /usr/bin/socat: no such file or directory: unknown"

**Pull request closed without merging**